docs/code/CrossPlatform.md

```diff
@@ -0,0 +1,175 @@
+# Cross-Platform and Architecture design doc
+
+### Table of contents  <!-- omit in toc -->
+- [Cross-Platform and Architecture design doc](#cross-platform-and-architecture-design-doc)
+	- [1. Why cross-platform/architecture](#1-why-cross-platformarchitecture)
+	- [2. Current status](#2-current-status)
+		- [2.1 Problems](#21-problems)
+		- [2.2 Build](#22-build)
+		- [2.3 Managed Code](#23-managed-code)
+		- [2.4 Native Projects](#24-native-projects)
+		- [2.5 3rd Party Dependencies](#25-3rd-party-dependencies)
+	- [3 Possible Solutions](#3-possible-solutions)
+		- [3.1 Rewrite native components to work on other platforms](#31-rewrite-native-components-to-work-on-other-platforms)
+		- [3.2 Rewrite native components to be only managed code](#32-rewrite-native-components-to-be-only-managed-code)
+		- [3.3 Hybrid Between Finding Replacements and Software Fallbacks](#33-hybrid-between-finding-replacements-and-software-fallbacks)
+	- [My Suggestion](#my-suggestion)
+		- [Improving managed fallback through intrinsics](#improving-managed-fallback-through-intrinsics)
+		- [3rd Party Dependencies](#3rd-party-dependencies)
+		- [Helix](#helix)
+		- [Mobile Support](#mobile-support)
+		- [Support Grid](#support-grid)
+
+## 1. Why cross-platform/architecture
+ML.NET is an open-source machine learning framework which makes machine learning accessible to .NET developers with the same code that powers machine learning across many Microsoft products, including Power BI, Windows Defender, and Azure.
+
+ML.NET allows .NET developers to develop/train their own models and infuse custom machine learning into their applications using .NET.
+
+Currently, while .NET is able to run on many different platforms and architectures, ML.NET is only able to run on Windows, Mac, and Linux, either x86 or x64. This excludes many architectures such as Arm, Arm64, M1, and web assembly, places that .NET is currently able to run.
+
+The goal is to enable ML.NET to run everywhere that .NET itself is able to run.
```

Comment:
Are there secondary goals? Eg 
1. Do not regress ML.NET performance on platforms where it currently runs. 
2. Where possible, aim for convergence to a single implementation for all platforms. (These goals may be in conflict..)
3. Where possible, avoid taking on ownership of code that is not core to ML.NET or does not align with our expertise. For example, we would probably not want to maintain a large math library.

Reply:
Another: 
4. Optimize for progressive delivery to reduce risk. ie., a solution where we expand to one platform at a time, or with partial functionality, release, and continue. No big bang effort that succeds or fails together.

Reply:
5. Where possible, prefer that dependencies be supplied with the platform (possibly a package manager) rather than shipped as part of ML.NET. (this is a weaker goal, as usability is important, so maybe it's a non goal)